src/Microsoft.Azure.WebJobs.ServiceBus/Common/GenericBinder.cs

```diff
@@ -90,7 +90,59 @@ internal class GenericBinder
             {
                 // Default, assume a Poco
                 return new PocoTriggerArgumentBinding<TMessage, TTriggerValue>(hooks, converterManager, elementType);
-            }         
+            }
+        }
+
+        /// <summary>
+        /// Creates a binding that binds to an <see cref="IFlushCollector{T}"/> with the specified argument type. Allows for a binding
+        /// to be generated for any POCO type.
+        /// </summary>
+        /// <param name="parameter">The ParameterInfo for the binding</param>
+        /// <param name="collectorGenericType">The generic type that must implement <see cref="IFlushCollector{T}"/> and have a public constructor with at most one parameter.</param>
+        /// <param name="collectorGenericArgumentType">The generic argument type for the <see cref="IFlushCollector{T}"/></param>
+        /// <param name="converterManager">A converter manager to pass along.</param>        
+        /// <param name="contextBinder">A Func that returns the TContext to be used by the <see cref="IFlushCollector{T}"/></param>
+        /// <returns></returns>
+        public static IBinding BindGenericCollector<TContext>(ParameterInfo parameter, Type collectorGenericType, Type collectorGenericArgumentType,
+            IConverterManager converterManager, Func<string, TContext> contextBinder)
```

Comment:
Is this semantically the same as 'invokeStringBinder' parameter in the BindCollector instance below? If so, can we use the same name to stay consistent.